webapp-django/crashstats/crashstats/views.py

```diff
@@ -817,24 +893,293 @@ def append_row_blob(blob, labels):
 
-    for version in versions:
-        product_version = '%s:%s' % (product, version)
+
+@pass_default_context
+def crashes_per_user(request, default_context=None):
+    context = default_context or {}
+
+    context['products'] = context['currentproducts']['products']
+
+    platforms_api = models.Platforms()
+    platforms = platforms_api.get()
+    form_class = forms.DailyFormByVersion
+
+    date_range_types = ['report', 'build']
+    hang_types = ['any', 'crash', 'hang-p']
+
+    form = form_class(
+        context['currentversions'],
+        platforms,
+        data=request.GET,
+        date_range_types=date_range_types,
+        hang_types=hang_types,
+    )
+    if form.is_valid():
+        params = form.cleaned_data
+        params['product'] = params.pop('p')
+        params['versions'] = sorted(list(set(params.pop('v'))), reverse=True)
         try:
-            blob = data['totals'][product_version]
+            params['platforms'] = params.pop('os')
         except KeyError:
-            continue
-        append_row_blob(blob, totals_labels)
-    writer.writerow(row)
-    return response
+            params['platforms'] = None
+    else:
+        return http.HttpResponseBadRequest(str(form.errors))
+
+    if len(params['versions']) > 0:
+        context['version'] = params['versions'][0]
+
+    context['product'] = params['product']
+
+    if not params['versions']:
+        # need to pick the default featured ones
+        params['versions'] = [
+            version['version']
+            for version in context['currentversions']
+            if version['product'] == params['product'] and version['featured']
+        ]
+
+    context['available_versions'] = []
+    now = datetime.datetime.utcnow().date()
+    for version in context['currentversions']:
+        start_date = isodate.parse_date(version['start_date'])
+        end_date = isodate.parse_date(version['end_date'])
+        if (
+            params['product'] == version['product'] and
+            start_date <= now and
+            end_date >= now
+        ):
+            context['available_versions'].append(version['version'])
+
+    if not params.get('platforms'):
+        params['platforms'] = [
+            x['name'] for x in platforms if x.get('display')
+        ]
+    context['platforms'] = params.get('platforms')
+
+    end_date = params.get('date_end') or datetime.datetime.utcnow()
+    if isinstance(end_date, datetime.datetime):
+        end_date = end_date.date()
+    start_date = (params.get('date_start') or
+                  end_date - datetime.timedelta(weeks=2))
+    if isinstance(start_date, datetime.datetime):
+        start_date = start_date.date()
+
+    context['start_date'] = start_date.strftime('%Y-%m-%d')
+    context['end_date'] = end_date.strftime('%Y-%m-%d')
+
+    context['duration'] = abs((start_date - end_date).days)
+    context['dates'] = utils.daterange(start_date, end_date)
+
+    context['hang_type'] = params.get('hang_type') or 'any'
```

Comment:
No, that's different. I want it to be `'any'` if the `hang_type` is not set at all or is falsy.